File: PoketDex/app/src/main/java/pt/rfsfernandes/data/remote/DataSource.java
package pt.rfsfernandes.data.remote;

import pt.rfsfernandes.custom.Constants;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;


public class DataSource {

  private static final String BASE_URL = Constants.BASE_URL;

  private static final Retrofit retrofit = new Retrofit.Builder()
      .baseUrl(BASE_URL)
      .addConverterFactory(GsonConverterFactory.create())
      .build();

  private static PokemonService sPokemonService;

  public static PokemonService getPokemonService() {
    if (sPokemonService == null) {
      sPokemonService = retrofit.create(PokemonService.class);
    }
    return sPokemonService;
  }

}

File: PoketDex/app/src/main/java/pt/rfsfernandes/pocketdex/data/remote/DataSource.kt
package pt.rfsfernandes.pocketdex.data.remote

import pt.rfsfernandes.pocketdex.custom.Constants
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object DataSource {
    private const val BASE_URL = Constants.BASE_URL
    private val retrofit = Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    private var sPokemonService: PokemonService? = null
    val pokemonService: PokemonService?
        get() {
            if (sPokemonService == null) {
                sPokemonService = retrofit.create(PokemonService::class.java)
            }
            return sPokemonService
        }
}